### PR TITLE
fix ash echo

### PR DESCRIPTION
**Pull request by xiao-mang**
## 拉取/合并请求描述：(PR description)

[

#### 为什么提交这份PR (why to submit this PR)

The return value of ioctl for F_DUPFD_CLOEXEC has an error.

#### 你的解决方案是什么 (what is your solution)

Add case F_DUPFD_CLOEXEC, and return correct value.

#### 在什么测试环境下测试通过 (what is the test environment)

qemu aarch64
]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: components/dfs/dfs_v2/src/dfs_file.c [hunk 1]
```diff
@@ -29,6 +29,7 @@
 
 
 #define MAX_RW_COUNT 0xfffc0000
+#define F_DUPFD_CLOEXEC 1030
```
Comment (geniusgogo) [inline components/dfs/dfs_v2/src/dfs_file.c:32]: 不应该出现在这个地方

File: components/dfs/dfs_v2/src/dfs_file.c [hunk 2]
```diff
@@ -1061,6 +1061,9 @@ int dfs_file_fcntl(int fd, int cmd, unsigned long arg)
         case F_SETLK:
         case F_SETLKW:
```
Comment (geniusgogo) [inline components/dfs/dfs_v2/src/dfs_file.c:1062]: 参照这些宏定义的地方

File: components/libc/compilers/musl/fcntl.h [hunk 3]
```diff
@@ -21,4 +21,8 @@
 #define O_BINARY 0x10000
 #endif
 
+#ifndef F_DUPFD_CLOEXEC
```
Comment (geniusgogo) [inline components/libc/compilers/musl/fcntl.h:24]: musl库里本身就有无需重新定义

File: components/dfs/dfs_v2/src/dfs_file.c [hunk 4]
```diff
@@ -1061,6 +1062,9 @@ int dfs_file_fcntl(int fd, int cmd, unsigned long arg)
...
             break;
+        case F_DUPFD_CLOEXEC:
```
Comment (geniusgogo) [inline components/dfs/dfs_v2/src/dfs_file.c:1065]: 只对SMART生效就好了

**Review by geniusgogo — CHANGES_REQUESTED** [review]

**Review by geniusgogo — APPROVED** [review]

**Review by BernardXiong — APPROVED** [review]

**Pull request merged**